pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/InMemoryDelayedDeliveryTracker.java

```diff
@@ -78,6 +83,17 @@ public boolean addMessage(long ledgerId, long entryId, long deliveryAt) {
             return false;
         }
 
+        synchronized (this) {
+            if (ComparisonChain.start().compare(ledgerId, lastLedgerId).compare(entryId, lastEntryId).result() <= 0) {
```

Comment:
Using Guava's ComparisonChain will create object instances.

```java
if (ledgerId < lastLedgerId || (ledgerId == lastLedgerId && entryId < lastEntryId))
```

Reply:
ok

Reply:
```
        if (ledgerId < lastLedgerId || (ledgerId == lastLedgerId && entryId <= lastEntryId)) {

```

changed to entryId <= lastEntryId  , if equals completely, we won't add

Reply:
Yes, correct.